File: 1.Multiples_of_3_and_5/1.Multiples_of_3_and_5.py
import sys

def Tn(n):
    return (n * (n + 1)) // 2

print("Please type the range:")
n = int(input().strip())
N = n - 1
nSum = 3 * Tn(N // 3) + 5 * Tn(N // 5) - 15 * Tn(N // 15)
print(nSum)

File: 1.Multiples_of_3_and_5/1.Multiples_of_3_and_5.hs
tn :: Int -> Int -> Int
tn fac n = fac * n' * (n' + 1) `div` 2 where n' = n `div` fac

three_or_five :: IO ()
three_or_five = do
    putStrLn "Please type the range:"
    n <- readLn
    -- or t_temp <- getLine --
    let m = n - 1
    print $ tn 3 m + tn 5 m - tn 15 m